proteus/NonlinearSolvers.py

```diff
@@ -617,6 +618,102 @@ def solve(self,u,r=None,b=None,par_u=None,par_r=None):
         log("   Newton it %d norm(r) = %12.5e  \t\t norm(r)/(rtol*norm(r0)+atol) = %12.5e"
             % (self.its,self.norm_r,(self.norm_r/(self.rtol_r*self.norm_r0+self.atol_r))),level=1)
 
+class POD_Newton(Newton):
+    """Newton's method on the reduced order system based on POD"""
+    def __init__(self,
+                 linearSolver,
+                 F,J=None,du=None,par_du=None,
+                 rtol_r  = 1.0e-4,
+                 atol_r  = 1.0e-16,
+                 rtol_du = 1.0e-4,
+                 atol_du = 1.0e-16,
+                 maxIts  = 100,
+                 norm = l2Norm,
+                 convergenceTest = 'r',
+                 computeRates = True,
+                 printInfo = True,
+                 fullNewton=True,
+                 directSolver=False,
+                 EWtol=True,
+                 maxLSits = 100):
+        Newton.__init__(self,
+                linearSolver,
+                F,J,du,par_du,
+                rtol_r,
+                atol_r,
+                rtol_du,
+                atol_du,
+                maxIts,
+                norm,
+                convergenceTest,
+                computeRates,
+                printInfo,
+                fullNewton,
+                directSolver,
+                EWtol,
+                maxLSits)
+        self.DB = 11
+        U = np.loadtxt('SVD_basis')
```

Comment:
@cekees sorry I missed these comments. I thought I had my gmail open on my laptop and just pushed some more additions. Yes, you should be able to build the rho and PF here. I added archiving for the 'mass' residual in Transport and NumericalSolution. That way, we can build the deim interpolants for both the full residual. Presumably, the rho indices built for the spatial term would do fine for the temporal terms here because they're linear but I think we need to have separate bases in general.